### ISSUE TITLE
onActive not called if timeout changes

### ISSUE THREAD

**Issue by VolodymyrSeredaBBC** [SupremeTechnopriest/react-idle-timer#143]
Hi! I think I discovered a bug with the IdleTimer.

Use-case: Start with one timeout, but change the timeout after the user becomes idle.

Example code illustrating the problem:

```
import React, { useState } from "react";
import ReactDOM from "react-dom";
import IdleTimer from "react-idle-timer";

const Component = (props) => {
  const [text, setText] = useState("initial");
  const [timeout, setTimeout] = useState(5000);

  return (
    <>
      <IdleTimer
        onActive={() => setText("active")}
        onIdle={() => {
          setText("idle");
          // This breaks the IdleTimer
          setTimeout(4000);
        }}
        timeout={timeout}
      />
      <div>{text}</div>
    </>
  );
};

ReactDOM.render(
  <React.StrictMode>
    <Component />
  </React.StrictMode>,
  document.getElementById("root")
);
```

Expected result:
1. Text is "initial"
2. After 5 seconds, text becomes "idle"
3. I interact with the page
4. Text immediately becomes "active"

Actual result:
1. Text is "initial"
2. After 5 seconds, text becomes "idle"
3. I interact with the page
4. Text remains "idle". **Interaction is ignored**.

This issue disappears if I remove the line `setTimeout(4000);`.

**Comment by SupremeTechnopriest** [comment]
If you are using functional components you might want to consider using the `useIdleTimer` hook.  So when you change the timeout, the timer is automatically reset.  I will add a test for this case and see why its not working as intended.

**Comment by SupremeTechnopriest** [comment]
Hey @VolodymyrSeredaBBC, I found the issue and patched it for both `IdleTimer` and `useIdleTimer`.  I added some tests for this case as well.  It has been published to npm as version `4.4.5`.  If for some reason it doesnt solve your issue, feel free to reopen this.

**Comment by VolodymyrSeredaBBC** [comment]
Thank you! That was very fast. The issue seems to be solved now :)
